network/connection.go

```diff
@@ -615,8 +615,11 @@ func (conn *Connection) packetHandler() {
 		} else {
 			defaultFirewallHandler(conn, pkt)
 		}
+
 		// log verdict
 		log.Tracer(pkt.Ctx()).Infof("filter: connection %s %s: %s", conn, conn.Verdict.Verb(), conn.Reason.Msg)
+		// submit trace logs
+		log.Tracer(pkt.Ctx()).Submit()
```

Comment:
I guess this could be `defer`ed

Reply:
Submitting itself is not expensive and must happen earlier so that the logs are in the correct order. Before, the SPN integration would log before the connection got accepted.